libtwolame/Makefile.am

```diff
@@ -3,6 +3,7 @@ AM_CFLAGS = -I ../build/ $(WARNING_CFLAGS)
 lib_LTLIBRARIES = libtwolame.la
 include_HEADERS = twolame.h
 
+libtwolame_la_CPPFLAGS = -DTWOLAME_BUILD
```

Comment:
Having to add this seems weird to me.
Probably not actually a problem but it also gets set for all OSes, not just Windows.

Is there a different way of doing this?

Reply:
if you look carefully, TWOLAME_BUILD is used only in a #ifdef _WIN32 , so anyway, TWOLAME_BUILD will be pased to the preprocessor only if a Windows compiler is used.
I can add a test in Makefile.am, but the test is already done with the #ifdef _WIN32, so it would be useless.

Reply:
https://www.gnu.org/software/libtool/manual/libtool.html#Windows-DLLs  :  

"When the objects that form the library are built, there are generally two copies built for each object. One copy is used when linking the DLL and one copy is used for the static library. On Windows systems, a pair of defines are commonly used to discriminate how the interface symbols should be decorated. The first define is ‘-DDLL_EXPORT’, which is automatically provided by Libtool when libtool builds the copy of the object that is destined for the DLL. The second define is ‘-DLIBFOO_BUILD’ (or similar), which is often added by the package providing the library and is used when building the library, but not when consuming the library. "

Reply:
Ah, thanks! It is really useful to know that is the official libtool way of doing it. I will try and get this merged tomorrow.

Reply:
another good reading : https://gcc.gnu.org/wiki/Visibility section "How to use the new C++ visibility support"